docs/markdown/snippets/language args in project are deprecated.md

```diff
@@ -0,0 +1,17 @@
+## Using language arguments in project(default_options) is deprecated
+
+One can place any valid command line option in the `project(default_options)`
+field. The problem is that some of these options result in very strange behavior.
+Case an point is c_args (and friends), which will do the following when placed
+in the project() options:
+
+1. prevent $CFLAGS from being read
+2. prevent the `c_args` from being read from a machine file (cross or native)
```

Comment:
Is that true? IIRC machine file override default_options.

Reply:
Default _options are == to command lines options I think. I will double check on Monday

Reply:
I gave it a try: cli > native file > default_options > env.

Reply:
arguably we should make it cli > native file > env > default_options. But better to not allow default_options for anything that take from env to avoid confusion IMHO. This probably also applies to e.g. pkg_config_path option.